src/ansys/units/dimensions.py

```diff
@@ -105,30 +104,21 @@ def __truediv__(self, other):
                 results[dim] = -value
         return Dimensions(results)
 
-    def __pow__(self, __value):
+    def __pow__(self, __value: float) -> Dimensions:
         results = self._dimensions.copy()
         for item in self:
             results[item[0]] *= __value
         return Dimensions(results)
 
-    def __eq__(self, __value):
-        dims = __value._dimensions.copy()
-        for dim, value in self:
-            if dim in dims:
-                dims[dim] -= value
-            else:
-                return False
-        if [False for v in dims.values() if v != 0]:
-            return False
-        return True
-
-    def __ne__(self, other):
-        return not self.__eq__(other)
+    def __eq__(self, __value: object) -> bool:
+        return (
+            isinstance(__value, Dimensions) and self._dimensions == __value._dimensions
+        )
```

Comment:
This seemed overcomplicated, maybe I'm missing something wrt to 0 handling

Reply:
Considering the title and the linked issue here, I would recommend keeping the relative changes only in this PR. It helps to review the changes. Do we need to update this __eq__ method for supporting common units?